### PR TITLE
chores: add chromatic

### PR DESCRIPTION
**Pull request by thiyagaraj-eightfold**
Add chromatic to the project

### CODE REVIEW

**Review by dkilgore-eightfold — APPROVED** [review]

**Review by ychhabra-eightfold — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -44,3 +44,13 @@ jobs:
           key: node_modules-${{ hashFiles('yarn.lock') }}
       - run: yarn
       - run: yarn lint
+  chromatic-deployment:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - name: Install dependencies
```
[the remark this answers is not in the document]
Comment (ychhabra-eightfold) [reply .github/workflows/build.yml:52]: yarn install also needs the correct node version

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -44,3 +44,15 @@ jobs:
           key: node_modules-${{ hashFiles('yarn.lock') }}
       - run: yarn
       - run: yarn lint
+  chromatic-deployment:
```
Comment (ychhabra-eightfold) [inline .github/workflows/build.yml:47]: We shouldn't be running this on the pull_request event but only on push event as recommend by the docs see https://www.chromatic.com/docs/github-actions#recommended-configuration-for-build-events

File: .github/workflows/chromatic.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
+name: 'Chromatic'
+
+on: push
+
+env:
+  NODE_VERSION: 16
+jobs:
+  chromatic-deployment:
+    runs-on: ubuntu-latest
```
Comment (ychhabra-eightfold) [inline .github/workflows/chromatic.yml:9]: No need to create another worklow yaml just add the following in the previous change you had done
```suggestion
    runs-on: ubuntu-latest
    if: github.event_name == 'push'
```

File: .github/workflows/chromatic.yml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    steps:
+      - uses: actions/checkout@v3
+        with:
+            fetch-depth: 0
+      - uses: actions/setup-node@v3
+      - name: Load cached node_modules
+        uses: actions/cache@v3
+        run: yarn
+        with:
```
Comment (ychhabra-eightfold) [inline .github/workflows/chromatic.yml:18]: actions cache doesnt need node_version

File: .github/workflows/chromatic.yml (hunk 3 above)
Comment (thiyagaraj-eightfold) [reply .github/workflows/chromatic.yml:9]: Want to keep it separate, else it just becomes unmanageable yaml

File: .github/workflows/chromatic.yml (hunk 4 above)
Comment (thiyagaraj-eightfold) [reply .github/workflows/chromatic.yml:18]: its for yarn you mentioned

Comment (ychhabra-eightfold) [reply .github/workflows/chromatic.yml:20]: i meant yarn needs 
- uses: actions/setup-node@v3
   with:
       node-version: ${{ env.NODE_VERSION }}

File: .github/workflows/chromatic.yml [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+          path: node_modules
+          key: node_modules-${{ hashFiles('yarn.lock') }}
+      - name: Publish to Chromatic
+        uses: chromaui/action@v1
+        with:
+          node-version: ${{ env.NODE_VERSION }}
```
Comment (ychhabra-eightfold) [inline .github/workflows/chromatic.yml:26]: dont think this is needed

**Comment by ychhabra-eightfold** [comment]
the workflow was not correct, you shouldnt merge without approvals @thiyagaraj-eightfold

**Pull request merged**